routing/src/main/kotlin/ru/surfstudio/compose/routing/NavigationDispatcher.kt

```diff
@@ -196,7 +196,7 @@ class NavigationDispatcher(
             if (controller.previousBackStackEntry?.destination == null) {
                 destinationPathList.clear()
             }
-            if (isBack && destinationPathList.isNotEmpty()) {
+            if (isBack && destinationPathList.isNotEmpty() && destinationPathList.size > 1) {
```

Comment:
```suggestion
            if (isBack && destinationPathList.size > 1) {
```
isNotEmpty check can be omitted